### ISSUE TITLE
Terminal has rendering issues with default font on Fedora

### ISSUE THREAD

**Issue by mbien** [apache/netbeans#5980]
### Apache NetBeans version

Apache NetBeans 17 and 18

### What happened

"Monospaced 15" which is picked by default on fedora has some rendering issues:
![image](https://github.com/apache/netbeans/assets/114367/aebcae18-7f36-49e8-8209-cf8a3e61b321)

other fonts like for example "Liberation Mono 15" seem to be working fine:
![image](https://github.com/apache/netbeans/assets/114367/47552539-88c6-4f21-b9f7-47f439ac4eb7)

### How to reproduce

run NB with fresh config and open the terminal

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Fedora 38

### JDK

tested on 17 (RedHat via package manager) and on 20 (Corretto)

### Apache NetBeans packaging

Apache NetBeans binary zip

### Are you willing to submit a pull request?

Maybe

**Comment by mbien** [comment]
If I remember correctly, `Monospaced` is actually a placeholder for the OS or the JDK to pick some monospaced font, which in this case seems to not give desirable results. I am wondering why though, This is either a really weird font or the picked font has indeed rendering issues.

**Comment by neilcsmith-net** [comment]
Can replicate with my Fedora test install in Virtualbox.

Curiously, as far as I can tell, the output window uses the same font, but displays OK.  The previews of both under `Options / Miscellaneous` show the same differences.

**Comment by mbien** [comment]
@eirikbakke have you seen something like this before? Rendering default fonts with broken kerning?

**Comment by neilcsmith-net** [comment]
@mbien I started taking a look at this, but got tied up with release stuff. I think it's possibly caused by this class? https://github.com/apache/netbeans/blob/master/ide/lib.terminalemulator/src/org/netbeans/lib/terminalemulator/MyFontMetrics.java

I noticed that `a` is one of few characters that has single spacing. The metrics is based on that character. It looks like the metrics rounds up any character that is slightly wider to double spacing. This class should possibly have a little leeway in it rather than expecting exact pixel size?
